lib/file.js

```diff
@@ -206,54 +205,54 @@ class File extends EventEmitter {
       if (!end) end = response.s
       if (start > end) return stream.emit('error', Error("You can't download past the end of the file."))
 
-      let activeStreams = 0
-      let currentOffset = apiStart
-      let chunkSize = initialChunkSize
-      let combined
-
-      function handleErrors (resp) {
-        if (resp.statusCode === 200) return
+      function handleMegaErrors (resp) {
+        if (resp.statusCode === 200) return;
         stream.emit('error', resp.statusCode === 509
         ? Error('Bandwidth limit reached: ' + resp.headers['x-mega-time-left'] + ' seconds until it resets')
-        : Error('MEGA returned a ' + resp.statusCode + ' status code'))
-        gotError = true
+        : Error('MEGA returned a ' + resp.statusCode + ' status code'));
       }
-
-      function getChunk () {
-        if (gotError) return
-        const currentMax = Math.min(end, currentOffset + chunkSize)
-        if (currentMax <= currentOffset) return
-        const r = requestModule(response.g + '/' + currentOffset + '-' + (currentMax - 1))
-
-        r.on('end', getChunk)
-        r.on('error', getChunk)
-        r.on('response', handleErrors)
-
-        combined.append(r)
-
-        currentOffset = currentMax
-        if (chunkSize < maxChunkSize) {
-          chunkSize = chunkSize + chunkSizeIncrement
-        }
-
-        activeStreams += 1
-        if (activeStreams < maxConnections) {
-          setTimeout(getChunk, 1000)
-        }
+      
+      function handleConnectionErrors(resp) {
```

Comment:
The function receives [an error](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Error), not [a response](https://nodejs.org/api/http.html#http_class_http_incomingmessage).